sql/stats.go

```diff
@@ -73,12 +74,75 @@ func (s *Stats) String() string {
 type Histogram []Bucket
 
 type Bucket struct {
-	Count      uint64        `json:"count"`
-	Distinct   uint64        `json:"distinct"`
-	BoundCount uint64        `json:"bound_count"`
-	Mcv        []interface{} `json:"mcv"`
-	McvCount   []uint64      `json:"mvc_count"`
-	UpperBound interface{}   `json:"upper_bound"`
+	Count      uint64          `json:"count"`
+	Distinct   uint64          `json:"distinct"`
+	Null       uint64          `json:"null"`
+	Mcv        [][]interface{} `json:"mcv"`
+	McvCount   []uint64        `json:"mvc_count"`
+	BoundCount uint64          `json:"bound_count"`
+	UpperBound []interface{}   `json:"upper_bound"`
+}
+
+func (b Bucket) MarshalJSON() ([]byte, error) {
+	return json.Marshal(&struct {
+		Count      uint64
+		Distinct   uint64
+		Null       uint64
+		Mcv        [][]interface{}
+		McvCount   []uint64
+		BoundCount uint64
+		UpperBound []interface{}
+	}{
+		Count:      b.Count,
+		Distinct:   b.Distinct,
+		Null:       b.Null,
+		UpperBound: b.UpperBound,
+		BoundCount: b.BoundCount,
+		Mcv:        b.Mcv,
+		McvCount:   b.McvCount,
+	})
+}
+
+func (b Bucket) String() string {
```

Comment:
(super minor) Out of curiosity again... do we need to format the string output this closely? Just wondering if we used `fmt.Sprintf("%v", b)` if we'd get something close enough and wouldn't have to remember to update this function when we change the struct.

Reply:
Maybe, I haven't figured out serializability yet. I think these objects should probably implement the `types.JSONValue` when its usage is fixed. And I think probably this will be an interface with a separate memory/Dolt implementation. It will look very different by the time it's usable in prod.